R/conditions.R

```diff
@@ -51,7 +51,7 @@ with_handlers <- function(code, handlers) {
 }
 
 sanitize_call <- function(cnd) {
-  if (identical(cnd$call, quote(eval(expr, envir)))) {
+  if (identical(cnd$call, quote(withVisible(do)))) {
```

Comment:
Is this the sanitation code that should trigger for knitr for the error ? 

I have a hard time for now understanding what `do` will be here. When is the identical supposed to be `TRUE` ? 

I am under the impression that this is not the expected value when ran inside **knitr** evaluation call, and why the sanitation may not happen.  When doing `knitr::knit()`, we definitely go into this function. 

Also not using **knitr** is throwing the same error not sanitized 
````
> evaluate::evaluate("mpg")
<evaluation>
Source code: 
  mpg
Condition: 
  Error in eval(as.call(list(cb)), envir):
  objet 'mpg' introuvable
````

I feel I am missing something here, or this is currently not sanitizing as we would like